src/Constraints/ArrayConstraint.php

```diff
@@ -26,8 +26,7 @@ public static function check(mixed $input, string $pointer, bool $coerce): vec<m
 
     $spec = TypeSpec\vec(TypeSpec\mixed());
     try {
-      # To allow for either PHP or hack arrays, we coerce to a vec here.
-      return $spec->coerceType($input);
+      return $spec->assertType($input);
     } catch (TypeCoercionException $e) {
```

Comment:
We'll want to update this, I think it's `TypeAssertionException`

Reply:
Also, would be great to add a test case here: https://github.com/slackhq/hack-json-schema/blob/master/tests/ArraySchemaValidatorTest.php for the case that this was failing before (ie. passing a dict but being considered valid)

Reply:
thanks! will update! i'm looking into whether i need to update tests rn too

Reply:
oh just saw your 2nd comment, sorry - created the PR but didn't think you'd see it until I posted it somewhere! yes, i'll add a test case!